File: filepicker/src/main/java/droidninja/filepicker/models/sort/NameComparator.java
package droidninja.filepicker.models.sort;

import java.util.Comparator;

import droidninja.filepicker.models.Document;

/**
 * Created by gabriel on 10/2/17.
 */

public class NameComparator implements Comparator<Document> {

    protected NameComparator() { }

    @Override
    public int compare(Document o1, Document o2) {
        return o1.getTitle().toLowerCase().compareTo(o2.getTitle().toLowerCase());
    }
}

File: filepicker/src/main/java/droidninja/filepicker/models/sort/NameComparator.kt
package droidninja.filepicker.models.sort

import droidninja.filepicker.models.Document
import java.util.*

/**
 * Created by gabriel on 10/2/17.
 */
class NameComparator : Comparator<Document> {
    override fun compare(o1: Document, o2: Document): Int {
        return o1.name.toLowerCase(Locale.getDefault()).compareTo(o2.name.toLowerCase(Locale.getDefault()))
    }
}